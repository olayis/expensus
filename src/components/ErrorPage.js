import React from 'react';
import { history } from '../routers/AppRouter';
import engineer from '../../public/img/QA-Engineer.svg';

const ErrorPage = () => (
  <div className='content-container'>
    <div className='error'>
      <img
        src={engineer}
        alt='A Quality Assurance Engineer'
        className='error__illustration'
        width='1075'
        height='585'
      />
      <h2 className='error__text'>Oops! Something went wrong.</h2>
      <div className='error__explanation'>
        Click here to try again.
        <span className='error__action'>
          <button className='btn btn--try-again' onClick={() => history.go(0)}>
            Try Again
          </button>
        </span>
      </div>
    </div>
  </div>
);

export default ErrorPage;
