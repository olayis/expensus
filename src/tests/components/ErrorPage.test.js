import React from 'react';
import { shallow } from 'enzyme';
import ErrorPage from '../../components/ErrorPage';

test('should render ErrorPage correctly', () => {
  const wrapper = shallow(<ErrorPage />);
  expect(wrapper).toMatchSnapshot();
});
