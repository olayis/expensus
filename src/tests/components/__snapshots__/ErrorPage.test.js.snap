// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ErrorPage correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="error"
  >
    <img
      alt="A Quality Assurance Engineer"
      className="error__illustration"
      height="585"
      src="SvgrURL"
      width="1075"
    />
    <h2
      className="error__text"
    >
      Oops! Something went wrong.
    </h2>
    <div
      className="error__explanation"
    >
      Click here to try again.
      <span
        className="error__action"
      >
        <button
          className="btn btn--try-again"
          onClick={[Function]}
        >
          Try Again
        </button>
      </span>
    </div>
  </div>
</div>
`;
